builder/tart/step_run.go

```diff
@@ -100,11 +101,14 @@ func (u uiWriter) Write(p []byte) (n int, err error) {
 func (s *stepRun) Cleanup(state multistep.StateBag) {
 	config := state.Get("config").(*Config)
 	ui := state.Get("ui").(packersdk.Ui)
+	cmd := state.Get("tart-cmd").(*exec.Cmd)
+	if cmd == nil {
+		return // Nothing to shut down
+	}
+	ui.Say("Gracefully shutting down the VM...")
 
 	communicator := state.Get("communicator")
 	if communicator != nil {
-		ui.Say("Gracefully shutting down the VM...")
```

Comment:
Perhaps this one shouldn't be moved after all, since sending `SIGINT` is not technically a graceful shutdown at the moment.

We could add another `ui.Say()` in the `else` arm of `if communicator != nil` below that says something useful about sending a signal to `tart run`.

Reply:
Good point. I'll add a feature to Tart to hook ` SIGUSR2` to `requestStop`, so that we can provide some more gracefull logic here. In the meantime, I'll change the message.

Reply:
>  I'll add a feature to Tart to hook  SIGUSR2 to requestStop, so that we can provide some more gracefull logic here. In the meantime, I'll change the message.

Unfortunately this just brings up the "Are you sure" dialog, so not very useful for automated gracefull shutdown, which I guess is why the SSH communicator is preferred if we have it.

<img width="1136" alt="image" src="https://github.com/cirruslabs/packer-plugin-tart/assets/7536/a8df0445-aaa4-4b31-9ec7-1d7bd0e905c3">

Still, might be useful on Linux, or if macOS can be told to not present the dialog: https://github.com/cirruslabs/tart/pull/842